db-server/test/backend/db_tests.js

```diff
@@ -54,1938 +54,1313 @@ function createEmail(data) {
+              assert.equal(err.error, 'Conflict', 'error')
+            })
         })
-        .then(function() {
-          return db.checkPassword(ACCOUNT.uid, {verifyHash: Buffer(crypto.randomBytes(32))})
+      })
+
+      describe('db.account', () => {
+        it('should return account', () => {
+          return db.account(accountData.uid)
+            .then((account) => {
+              assert.deepEqual(account.uid, accountData.uid, 'uid')
+              assert.equal(account.email, accountData.email, 'email')
+              assert.deepEqual(account.emailCode, accountData.emailCode, 'emailCode')
+              assert.equal(!! account.emailVerified, accountData.emailVerified, 'emailVerified')
+              assert.deepEqual(account.kA, accountData.kA, 'kA')
+              assert.deepEqual(account.wrapWrapKb, accountData.wrapWrapKb, 'wrapWrapKb')
+              assert(! account.verifyHash, 'verifyHash field should be absent')
+              assert.deepEqual(account.authSalt, accountData.authSalt, 'authSalt')
+              assert.equal(account.verifierVersion, accountData.verifierVersion, 'verifierVersion')
+              assert.equal(account.createdAt, accountData.createdAt, 'createdAt')
+              assert.equal(account.verifierSetAt, accountData.createdAt, 'verifierSetAt has been set to the same as createdAt')
+              assert.equal(account.locale, accountData.locale, 'locale')
+            })
         })
-        .then(function() {
-          assert(false, 'password check should fail')
-        }, function(err) {
-          assert(err, 'incorrect password produces an error')
-          assert.equal(err.code, 400, 'error code')
-          assert.equal(err.errno, 103, 'error errno')
-          assert.equal(err.message, 'Incorrect password', 'message')
-          assert.equal(err.error, 'Bad request', 'error')
+      })
+
+      describe('db.emailRecord', () => {
+        it('should return email record', () => {
+          return db.emailRecord(accountData.emailBuffer)
+            .then((account) => {
+              assert.deepEqual(account.uid, accountData.uid, 'uid')
+              assert.equal(account.email, accountData.email, 'email')
+              assert.deepEqual(account.emailCode, accountData.emailCode, 'emailCode')
+              assert.equal(!! account.emailVerified, accountData.emailVerified, 'emailVerified')
+              assert.deepEqual(account.kA, accountData.kA, 'kA')
+              assert.deepEqual(account.wrapWrapKb, accountData.wrapWrapKb, 'wrapWrapKb')
+              assert(! account.verifyHash, 'verifyHash field should be absent')
+              assert.deepEqual(account.authSalt, accountData.authSalt, 'authSalt')
+              assert.equal(account.verifierVersion, accountData.verifierVersion, 'verifierVersion')
+              assert.equal(account.verifierSetAt, accountData.verifierSetAt, 'verifierSetAt')
+              // locale not returned with .emailRecord() (unlike .account() when it is)
```

Comment:
Rather than a comment saying this, would an assertion that proves it be more useful?